server/routes/profile.go

```diff
@@ -172,6 +174,73 @@ func EditProfile(w http.ResponseWriter, r *http.Request) {
 }
 
 func ProfilePic(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusAccepted)
-	w.Write([]byte("called pic"))
+	//Connect to datastore
+	ctx := appengine.NewContext(r)
+	dsClient := services.Locator.DsClient()
+
+	//Retrieve user from datastore
+	vars := mux.Vars(r)
+	paramID := vars["id"]
+	userService := services.Locator.UserService()
+	userID := userService.GetCurrentUserID(ctx)
+	if len(userID) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("no ID provided"))
+		return
+	}
+	if userID != paramID {
```

Comment:
I don't think you need the `paramID` at all

Reply:
Btw if you think sending a PATCH to `/user/` is strange, Facebook have invented a good, in my opinion, solution: use `/user/me/`, so you don't have to specify an id when you're referring to yourself